kDriveCore/BackgroundTasksService.swift

```diff
@@ -0,0 +1,143 @@
+ OR
+  e -l objc -- (void)[[BGTaskScheduler sharedScheduler] _simulateExpirationForTaskWithIdentifier:@"com.infomaniak.background.long-refresh"]
+
+  Force early termination ->
+  e -l objc -- (void)[[BGTaskScheduler sharedScheduler] _simulateExpirationForTaskWithIdentifier:@"com.infomaniak.background.refresh"]
+ OR
+  e -l objc -- (void)[[BGTaskScheduler sharedScheduler] _simulateExpirationForTaskWithIdentifier:@"com.infomaniak.background.long-refresh"]
+ */
+
+/// Service to ask the system to do some work in the background later.
+public protocol BackgroundTasksServiceable {
+    /// Ask the system to handle the app's background refresh
+    func registerBackgroundTasks()
+
+    /// Schedule next refresh with the system
+    func scheduleBackgroundRefresh()
+}
+
+struct BackgroundTasksService: BackgroundTasksServiceable {
+    @LazyInjectService private var scheduler: BGTaskScheduler
+    @LazyInjectService private var accountManager: AccountManageable
+    @LazyInjectService private var uploadQueue: UploadQueue
+
+    public init() {
+        // META: keep SonarCloud happy
+    }
+
+    public func registerBackgroundTasks() {
+        Log.backgroundTaskScheduling("registerBackgroundTasks")
+        registerBackgroundTask(identifier: Constants.backgroundRefreshIdentifier)
+        registerBackgroundTask(identifier: Constants.longBackgroundRefreshIdentifier)
+    }
+
+    public func buildBackgroundTask(_ task: BGTask, identifier: String) {
+        scheduleBackgroundRefresh()
+
+        handleBackgroundRefresh { _ in
+            Log.backgroundTaskScheduling("Task \(identifier) completed with SUCCESS")
+            task.setTaskCompleted(success: true)
+        }
+
+        task.expirationHandler = {
+            Log.backgroundTaskScheduling("Task \(identifier) EXPIRED", level: .error)
+            uploadQueue.suspendAllOperations()
+            uploadQueue.rescheduleRunningOperations()
+            task.setTaskCompleted(success: false)
+        }
+    }
+
+    func registerBackgroundTask(identifier: String) {
+        let registered = scheduler.register(
+            forTaskWithIdentifier: identifier,
+            using: nil
+        ) { task in
+            buildBackgroundTask(task, identifier: identifier)
+        }
+        Log.backgroundTaskScheduling("Task \(identifier) registered ? \(registered)")
+    }
+
+    func handleBackgroundRefresh(completion: @escaping (Bool) -> Void) {
+        Log.backgroundTaskScheduling("handleBackgroundRefresh")
+        // User installed the app but never logged in
+        if accountManager.accounts.isEmpty {
+            completion(false)
+            return
+        }
+
+        Log.backgroundTaskScheduling("Enqueue new pictures")
+        @InjectService var photoUploader: PhotoLibraryUploader
+        photoUploader.scheduleNewPicturesForUpload()
+
+        Log.backgroundTaskScheduling("Clean errors for all uploads")
+        @InjectService var uploadQueue: UploadQueue
```

Comment:
This can be removed now that it's @LazyInjectService

Reply:
`handleBackgroundRefresh` been called by the system and all, I was I too cautious ?
 Lazy is well tested and should work in any conditions, so yeah I'll move all the DI in lazy properties.